### PR TITLE
Create depoyment and update status on failure as well

### PR DESCRIPTION
**Pull request by baelter**
### WHY are these changes introduced?

Deploys fail silently

### WHAT is this pull request doing?

Always updating deployments, even on failure. Will create notifications in slack on failure.

### HOW can this pull request be tested?

Merge and try

### CODE REVIEW

**Comment by dentarg** [comment]
> ### HOW can this pull request be tested?
> Merge and try

It is possible to reference this branch, you can try with https://github.com/84codes/dev-playground84

File: .github/workflows/heroku.yml [hunk 1]
```diff
@@ -48,13 +38,25 @@ jobs:
...
     - name: Create deployment status in GitHub
+      if: always()
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
         gh api repos/${{ github.repository }}/deployments/$DEPLOYMENT_ID/statuses \
           --method POST \
           --header "Accept: application/vnd.github.v3+json" \
-          --field state=success \
+          --field state=${{ job.status }} \
```
Comment (dentarg) [inline .github/workflows/heroku.yml:60]: Can we use the status of the job before the job has finished?

I was thinking we would have to look at the conclusion of the step that pushes? https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context

**Comment by baelter** [comment]
Screenshot from 2023-09-22 15-33-39

File: .github/workflows/heroku.yml (hunk 1 above)
Comment (baelter) [reply .github/workflows/heroku.yml:60]: Yes, status is updated after each step

**Review by dentarg — APPROVED** [review]
Sweet

**Pull request merged**